Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -660,23 +690,29 @@ private extension PurchasesOrchestrator {
         }
         let unsyncedAttributes = self.unsyncedAttributes
 
-        backend.post(receiptData: receiptData,
-                     appUserID: appUserID,
-                     isRestore: allowSharingAppStoreAccount,
-                     productData: productData,
-                     presentedOfferingIdentifier: presentedOfferingID,
-                     observerMode: !finishTransactions,
-                     subscriberAttributes: unsyncedAttributes) { result in
+        self.backend.post(receiptData: receiptData,
+                          appUserID: appUserID,
+                          isRestore: allowSharingAppStoreAccount,
+                          productData: productData,
+                          presentedOfferingIdentifier: presentedOfferingID,
+                          observerMode: !finishTransactions,
+                          subscriberAttributes: unsyncedAttributes) { result in
             self.handleReceiptPost(withTransaction: transaction,
                                    result: result,
                                    subscriberAttributes: unsyncedAttributes)
         }
     }
 
-    func handleReceiptPost(withTransaction transaction: SKPaymentTransaction,
+    func handleReceiptPost(withTransaction transaction: StoreTransaction,
                            result: Result<CustomerInfo, BackendError>,
                            subscriberAttributes: SubscriberAttributeDict?) {
-        operationDispatcher.dispatchOnMainThread {
+        func finishTransaction() {
+            if self.finishTransactions, let sk1Transaction = transaction.sk1Transaction {
+                self.storeKitWrapper.finishTransaction(sk1Transaction)
```

Comment:
the RTTI here feels a little weird... why are we special casing SK1? [SK2 transactions have a finish](https://developer.apple.com/documentation/storekit/transaction) method too

Reply:
Yeah I agree this is messy. The reason is SK2 transactions are finished by [`StoreKit2TransactionListener`](https://github.com/RevenueCat/purchases-ios/blob/4.5.2/Sources/Purchasing/StoreKit2/StoreKit2TransactionListener.swift#L107), so I wanted to keep the existing behavior here. Even though this method uses any `StoreTransaction`, I didn't want to call that twice on SK2 transactions :/

But looking at this now, maybe we can remove the `finish` call in `StoreKit2TransactionListener` and call it here for consistency?

Reply:
I think that's probably a good call, although it doesn't need to be a part of this PR

Reply:
Okay let me make a Jira for it then.

Reply:
https://revenuecats.atlassian.net/browse/CSDK-122

This is ready for a final look then :)